src/main/webapp/dosave.jsp

```diff
@@ -279,172 +280,190 @@
+                                            && !request.getParameter(combinedFeldnamenArray[j] + "[" + i + "]").equals("-1")
+                                            && !request.getParameter(combinedFeldnamenArray[j] + "[" + i + "]").equals("NULL")) {
+                                        aenderung = true;
+                                    }
+                                }
+                                if (aenderung) {
+                                    String sql = "INSERT INTO " + zieltabelle + " SET " + formularAttribut + " = '" + id + "'";
+                                    for (int j = 0; j < combinedFeldnamenArray.length; j++) {
+                                        if (!combinedFeldnamenArray[j].equals("")) {
+                                            if (combinedFeldtypenArray[j].equals("textfield")
+                                                    || combinedFeldtypenArray[j].equals("textarea")
+                                                    || combinedFeldtypenArray[j].equals("select")
+                                                    || combinedFeldtypenArray[j].equals("sqlselect")
+                                                    || combinedFeldtypenArray[j].equals("addselect") || combinedFeldtypenArray[j].equals("addselectandtext")) {
+
+                                                String parameterValue = request.getParameter(combinedFeldnamenArray[j] + "[" + i + "]");
+                                                if (parameterValue != null) {
+                                                    if (parameterValue.equals("NULL")) {
+                                                        sql += ", " + zielattributArray[j] + " = NULL";
+                                                    } else {
+                                                        sql += ", " + zielattributArray[j] + " = '" + DBtoDB(parameterValue.trim()) + "'";
+                                                    }
+                                                }
+                                            } else if (combinedFeldtypenArray[j].equals("checkbox")) {
+                                                String checkboxValue = request.getParameter(combinedFeldnamenArray[j] + "[" + i + "]");
+                                                if (checkboxValue != null) {
+                                                    sql += ", " + zielattributArray[j] + " = '" + (checkboxValue.equals("on") ? "1" : "0") + "'";
+                                                }
                                             }
                                         }
                                     }
+                                    SaveHelper.insertOrUpdateSql(sql);
                                 }
-                                SaveHelper.insertOrUpdateSql(sql);
                             }
                         }
                     }
+                }catch (Exception e) {
```

Comment:
Den Text einer Exception zu prüfen und basierend darauf eine neue Exception zu werfen macht im Normalfall keinen Sinn, z.B. weil es unterschiedliche Arten von Exceptions gibt mit unterschiedlichen Bedeutungen die verschiedene  Ursachen haben können und man die nicht alle gleich behandeln darf, sonst kann sein man gibt die falsche  Meldung aus.

z.B. wenn ein Statement falsch ist und es eine Spalte in der DB nicht gibt, oder ein Update ausgeführt wird und ein Unique-Schlüssel falsch ist, und so weiter.

Hier ein paar Beispiele:
https://www.baeldung.com/hibernate-exceptions

ich denke alles sinnvoll abzufangen wäre viel zu komplex. Wir sollten uns also auf ein paar einfache Fälle beschränken. Die Meldung "ID nicht vorhanden" wäre dann vermutlich eher sowas wie die javax.persistence.EntityNotFoundException die man dann in eine IdNotFoundException umwandeln könnte. 

https://stackoverflow.com/questions/46612968/hibernate-unable-to-find-entity-with-id

Aber auch da => wahrscheinlich wäre es besser das im error.jsp zu machen statt im dosave.jsp. Falls im dosave  ein error auftaucht erfolgt das error handling ja sowieso im error.jsp.

Reply:
Das ist "zukomplex"  die else if abfrage startet bei zeile : 289 und geht bis 450,  meine lösung ist vielleicht nicht konvenktionell tut aber sehr gut,

Reply:
Ich merge das jetzt mal aber gehe davon aus dass an dieser Stelle früher oder später viele falsche Fehlermeldungen produziert werden und wir da über kurz oder lang noch nacharbeiten müssen.